cmd/imgcollector/image/hostfs/containerd_image.go

```diff
@@ -42,6 +42,16 @@ func NewContainerdImage(hash v1.Hash, cfg ContainerdHostFSConfig) (Image, error)
 		configBytes: configBytes,
 		contentDir:  cfg.ContentDir,
 	}
+
+	mi, err := manifestReader.resolveDigest()
```

Comment:
resolveDigest and resolveIndex looks redundand and reads manifest and index second time.

Inside manifestReader.resolveManifest() we already read index if it exists.

I suggest to change manifestReader to something more generic like metadataReader.resolve which returns struct containing both manifest and index.